### DISCUSSION TITLE
Título: Problema com retorno incorreto da API no curso Fullstack .NET

### DISCUSSION THREAD

**Discussion by renatotiburcio** [balta-io/forum#580] [Tira Dúvidas] [answered]
Estou enfrentando um problema com os endpoints que utilizam o RequireAuthorization na API do curso Fullstack .NET. Mesmo quando não estou logado, os endpoints são acessíveis e a aplicação tenta redirecionar para /Account/Login?ReturnUrl=endpoinorigem, resultando em um erro 404, pois o endereço não existe. Em vez disso, eu gostaria que a aplicação retornasse o código de status correto, como 401 (Unauthorized) ou 403 (Forbidden).

Como posso configurar a aplicação para que ela não redirecione para a página de login e, em vez disso, retorne o código de status apropriado?
Obs: Estou usando o projeto disponibilizado no github -->> https://github.com/balta-io/3054

![image](https://github.com/balta-io/forum/assets/46468715/9d2fc990-9e64-4d21-898b-7dc37c131970)

![image](https://github.com/balta-io/forum/assets/46468715/0e721c2a-533d-44fb-a155-dcfd46b578a8)

**Comment by renatotiburcio** [comment, marked as answer]
Resolvi deixando assim

public static void AddSecurity(this WebApplicationBuilder builder)
 {
     builder.Services
         .AddAuthentication(IdentityConstants.ApplicationScheme)
         .AddIdentityCookies();

     builder.Services.ConfigureApplicationCookie(options => {
         options.Events.OnRedirectToAccessDenied = context => {
             context.Response.StatusCode = 403;
             return Task.CompletedTask;
         };

         options.Events.OnRedirectToLogin = context => {
             context.Response.StatusCode = 401;
             return Task.CompletedTask;
         };
     });

     builder.Services.AddAuthorization();
 }

**Reply by andrebaltieri** [reply]
@renatotiburcio muito obrigado por compartilhar a solução 🚀
Existe a opção também de configurar as URLs também:
```csharp
options.AccessDeniedPath = "/acesso-negado/";
options.LoginPath = "/entrar/";
options.LogoutPath = "/sair/";
```
